zstash/globus.py

```diff
@@ -246,9 +264,45 @@ def globus_transfer(
         logger.error("Exception: {}".format(e))
         sys.exit(1)
 
+    # test for blocking
+    task_status = "UNKNOWN"
+    if not non_blocking:
+        logger.info(f"{ts_utc()}: BLOCKING START: invoking task_wait for task_id = {task_id}")
+        wait_timeout = 5
+        max_retries = 3
+        retry_count = 0
+        while retry_count < max_retries:
```

Comment:
I don't really understand this part. So, we just give up after `max_retries * wait_timeout` = `3*5` = `15` seconds? And pretend like we wanted to be non-blocking? Don't Globus transfers typically take hours?

Reply:
These values were made artificially small for test purposes (rather than incur massive test delays.)  You don's see it here, but in create.py I had changed MAXSIZE from n*GB to n*100,000 bytes, to force multiple tarfile generations quickly.

If you don't use a "while ... retry < max_retries" loop, and the transfer hangs, then zstash hangs.  You want to avoid that.

I'll provide more reasonable values here.

Reply:
Oh I see, that makes sense, thanks.